include/zephyr/bluetooth/hci_types.h

```diff
@@ -571,6 +571,46 @@ struct bt_hci_rp_read_tx_power_level {
 	int8_t   tx_power_level;
 } __packed;
 
+#define BT_HCI_LE_TX_POWER_PHY_1M               0x01
+#define BT_HCI_LE_TX_POWER_PHY_2M               0x02
+#define BT_HCI_LE_TX_POWER_PHY_CODED_S8         0x03
+#define BT_HCI_LE_TX_POWER_PHY_CODED_S2         0x04
+#define BT_HCI_OP_LE_ENH_READ_TX_POWER_LEVEL    BT_OP(BT_OGF_LE, 0x0076)
+struct bt_hci_cp_le_read_tx_power_level {
+	uint16_t handle;
+	uint8_t  phy;
+} __packed;
+
+struct bt_hci_rp_le_read_tx_power_level {
+	uint8_t  status;
+	uint16_t handle;
+	uint8_t  phy;
+	int8_t   current_tx_power_level;
+	int8_t   max_tx_power_level;
+} __packed;
+
+#define BT_HCI_OP_LE_READ_REMOTE_TX_POWER_LEVEL	BT_OP(BT_OGF_LE, 0x0077)
+
+#define BT_HCI_LE_TX_POWER_REPORT_DISABLE       0x00
+#define BT_HCI_LE_TX_POWER_REPORT_ENABLE        0x01
+#define BT_HCI_OP_LE_SET_TX_POWER_REPORT_ENABLE BT_OP(BT_OGF_LE, 0x007A)
+
+enum bt_conn_le_tx_power_report_reason {
+	/** Local Transmit power changed. */
+	BT_CONN_LE_TX_POWER_REPORT_REASON_LOCAL_CHANGED = 0,
+
+	/** Remote Transmit power changed. */
+	BT_CONN_LE_TX_POWER_REPORT_REASON_REMOTE_CHANGED = 1,
+
+	/** HCI_LE_Read_Remote_Transmit_Power_Level command completed. */
+	BT_CONN_LE_TX_POWER_REPORT_REASON_READ_REMOTE_COMPLETED = 2,
+};
```

Comment:
I guess this should be moved closer to `struct bt_hci_evt_le_transmit_power_report` since that's where you have `uint8_t reason`? Also, since it's `uint8_t reason` and not `enum bt_conn_le_tx_power_report_reason` I'd stick to defines, like everything else in this header file, and the prefix would then be `BT_HCI_*` instead of `BT_CONN_*` or `bt_conn_*` (if you check, there aren't any other definitions in this header file that don't use the bt_hci prefix, so already from that perspective it's ill-suited here)

Reply:
I see your points. Reason why I placed it there was as we technically only use it in conn.h and in connection with read remote tx power. 
How about changing the report type to `enum bt_conn_le_tx_power_report_reason` (I would rename it to `enum bt_hci_le_tx_power_report_reason` ), actually using that in `bt_hci_evt_le_transmit_power_report` (as then it is defined right above, would be kind-off ugly imo to not utilize it), and keep utilizing it in `bt_conn_le_tx_power_report` in conn.h? Yes we would have an hci type (`enum bt_hci_le_tx_power_report_reason`) in conn.h, but we reference the hci spec regarding reporting reason above that parameter anyways.

Reply:
See the new changes for how I meant this.